sample-operators/tomcat-operator/src/main/java/io/javaoperatorsdk/operator/sample/webapp/WebappReconciler.java

```diff
@@ -20,42 +19,26 @@
 import io.javaoperatorsdk.operator.api.reconciler.Context;
 import io.javaoperatorsdk.operator.api.reconciler.ControllerConfiguration;
 import io.javaoperatorsdk.operator.api.reconciler.DeleteControl;
-import io.javaoperatorsdk.operator.api.reconciler.EventSourceInitializer;
 import io.javaoperatorsdk.operator.api.reconciler.Reconciler;
 import io.javaoperatorsdk.operator.api.reconciler.UpdateControl;
-import io.javaoperatorsdk.operator.processing.event.ResourceID;
-import io.javaoperatorsdk.operator.processing.event.source.EventSourceRegistry;
-import io.javaoperatorsdk.operator.processing.event.source.InformerEventSource;
+import io.javaoperatorsdk.operator.sample.tomcat.resource.Tomcat;
+import io.javaoperatorsdk.operator.sample.webapp.resource.Webapp;
+import io.javaoperatorsdk.operator.sample.webapp.resource.WebappStatus;
 
 import okhttp3.Response;
 
 @ControllerConfiguration
-public class WebappReconciler implements Reconciler<Webapp>, EventSourceInitializer<Webapp> {
+public class WebappReconciler extends WebappEventSourceInitializer implements Reconciler<Webapp> {
```

Comment:
I'm kinda against inheritance in service layer, (it's harder to unit test test mainly in general). Can be make this an composition rather? SO implement the Initializer and maybe just instatiate a class there where you pass the event registry?

Reply:
Thanks for the nice comment!

That's right, it is advantageous for unit tests later to solve it by combination rather than inheritance.

I think what if we proceed with EventInitializer in Controller ?

To make that possible, we'll probably need some change in some APIs.

for example, users can inject their own custom event sources if they wish. Using the API like below.

```
final var operator = new Operator(client, DefaultConfigurationService.instance());
final List<EventSourceInitializer<?>> eventSourceInitialzers = ...

operator.register(reconciler, eventSourceInitialzers);
```

Then, `io.javaoperatorsdk.operator.processing.Controller` receives EventSourceInitializer parameter and passes `eventSourceManager` to process it.

With this improvement, I think that Reconciler can de-couple with EventSourceInitilizer.

Reply:
If we proceed with this improvement, it seems that we can make an issue separately and apply it to the sample code after processing.

If you agree with this change, I'll create a separate issue and send a pull-request again. :)

Reply:
Maybe with this improvement we can get rid of this code?

https://github.com/java-operator-sdk/java-operator-sdk/blob/efc0f058c6b0b93a06327a9f47aadb308b67dd18/operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/Controller.java#L171-L173

Reply:
You can just pass the registry to an another class currently. 

I agree that is not the nicest way have as it is currently, but note that usually you want to reference the instances of the event source from the reconciler to access the cache during the reconciliation. With just passing them separately we have to solve that part too. That why the current approach is kind a compromise.

Reply:
Another problem with this sample is that owner references in it. That could be fixes too :)

Reply:
Maybe discuss this problem on Thursday community meeting more in depth?

Reply:
I have an appointment this week, so it's difficult to attend, but I'll try to attend the next meeting. :) 

In this pull-request, I think about the idea and apply it so that it can be used in a composition way for de-coupling `reconciler` and `eventsoruce` :)

Reply:
Not sure if I follow the last sentence. So what I see here, is just extracting a method to a subclass, not sure if this has any strong benefit.

Reply:
yeq, ok First, it was changed to simply inject and use eventInitilzer. :)

Reply:
https://github.com/java-operator-sdk/java-operator-sdk/pull/722/commits/2d81fe9359264876649ff30a632caa0fc508ce61

However, this refactoring feels like a Delegate, since we currently have to implement EventSourceInitializer in our Reconciler.
 There don't seem to be any major strengths. so, we can also rollback this part to a previous version. 
 what do you think? 🤔

Reply:
yes, that is neither a very useful simplification, having just the method implemented as was originally is good enough with this architecture I think.

Reply:
I agree. I deleted the part separated by EventSourceInitilizer and rolled back to the originally way.